.github/mergify.yml

```diff
@@ -58,26 +143,307 @@ pull_request_rules:
+      comment:
+        message: |
+          > [!IMPORTANT]
+          > ## Description is necessary and should not be empty.
+          >
+          > Kindly provide details with **what** was changed, **why** it was changed.
 
   - name: "remove outdated reviews"
     conditions:
+      - and: *is_open
       - or: *default_branch
     actions:
       dismiss_reviews:
         changes_requested: true
         approved: true
         message: "This Pull Request has been updated, so we're dismissing all reviews."
 
-  - name: close stale pull request
+  - name: "remove triage label if approved"
+    conditions:
+      - and: *is_open
+      - '#approved-reviews-by>=1'
+    actions:
+      label:
+        remove:
+          - triage
+
+  - name: label inactive PRs as stale
+    conditions:
+      - and: *is_open
+      - commits[*].date_committer < 30 days ago
+    actions:
+      label:
+        toggle: ["stale"]
+
+  - name: close automated PRs with persistent merge conflicts quickly
+    conditions:
+      - and: *is_open
+      - or: *automated_prs
+      - "conflict"
+      - commits[*].date_committer < 1 days ago
+    actions:
+      close:
+        message: |
+          This automated PR has been closed due to merge conflicts.
+
+  - name: close stale PRs with merge conflicts
+    conditions:
+      - and: *is_open
+      - "conflict"
+      - commits[*].date_committer < 30 days ago
+    actions:
+      close:
+        message: |
+          This PR has been closed due to inactivity and merge conflicts.
+          Please resolve the conflicts and reopen if necessary.
+
+  - name: close stale pull request after 90 days
+    conditions:
+      - and: *is_open
+      - or: *default_branch
+      - commits[*].date_committer < 90 days ago
+    actions:
+      close:
+        message: |
+          This pull request looks stale. Feel free to reopen it if you think it's a mistake.
+
+  - name: close pull request waiting on feedback for 1 month
     conditions:
+      - and: *is_open
       - or: *default_branch
-      - -closed
+      - "label~=(feedback)"
       - updated-at < 30 days ago
     actions:
       close:
         message: |
-          This pull request looks stale. Feel free to reopen it if you think it's a mistake.
\ No newline at end of file
+          We haven't heard back from you, so we're closing this pull request.
+          Feel free to reopen it if you think it's a mistake.
+
+  - name: close pull request marked as invalid, duplicate or won't fix
+    conditions:
+      - and: *is_open
+      - or: *default_branch
+      - "label~=(duplicate|invalid|wontfix)"
+    actions:
+      close:
+        message: |
+          This pull request is no longer applicable.
+          Feel free to reopen it if you think it's a mistake.
+
+  - name: close pull request that is a work in progress and in active for 1 month
+    conditions:
+      - and: *is_open
+      - or: *default_branch
+      - "label~=(WIP|wip|do-not-merge|do not merge)"
```

Comment:
I think there could be some circumstances where 'do not merge' is used to wait for an upstream dependency to be changed, so perhaps these shouldn't be closed.

Reply:
I think then it just comes down to "how long"? It could be due to an upstream change, but at some point it's still invalid. E.g. "do not merge" for one year? Right now, I have it at 30 days.  Not hung up on the number of days, just that there should be some limit.

Reply:
Ah yes good point. Based on past experiences, I'd give WIP 30 days and do not merge 6 months.